benchmarks/babelstream/src/babelStreamCommon.hpp

```diff
@@ -0,0 +1,144 @@
+#pragma once
+#include <chrono>
+#include <cmath>
+#include <ctime>
+#include <iomanip>
+#include <limits>
+#include <map>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+// This is defined because the message of INFO macro of CATCH2 is only printed by "-s" command line option and "-s"
+// also prints REQUIRE checks of the test output and their results. This macro is actually WARN macro of Catch2 with a
+// different name. The message is directly printed as in WARN; and REQUIREs are not added to the test output.
+#define INFO_METADATA(msg)                                                                                            \
+    INTERNAL_CATCH_MSG("INFO_METADATA", Catch::ResultWas::Warning, Catch::ResultDisposition::ContinueOnFailure, msg)
+
+namespace
+{
+    [[maybe_unused]] constexpr auto arraySizeMain = 1024 * 1024;
+    // Mul and Triad kernel params
+    [[maybe_unused]] constexpr auto scalarVal = 2.0f;
+    // DotKernel test workdiv params
+    [[maybe_unused]] constexpr auto blockThreadExtentMain = 1024;
+
+    [[maybe_unused]] constexpr auto valA = 1.0f;
+
+    //!
+    //! \brief FuzzyEqual compares two floating point or integral type values
+    //!
+    template<typename T>
+    [[maybe_unused]] bool FuzzyEqual(T a, T b)
+    {
+        if constexpr(std::is_floating_point_v<T>)
+        {
+            return std::fabs(a - b) < std::numeric_limits<T>::epsilon() * static_cast<T>(100.0);
+        }
+        else if constexpr(std::is_integral_v<T>)
+        {
+            return a == b;
+        }
+        else
+        {
+            static_assert(
+                std::is_floating_point_v<T> || std::is_integral_v<T>,
+                "FuzzyEqual<T> is only supported for integral or floating-point types.");
+        }
+    }
+
+    //!
+    //! \brief getCurrentTimestamp gets the time-stamp and returns it as a string
+    //!
+    [[maybe_unused]] static std::string getCurrentTimestamp()
```

Comment:
Is this really required? Is see nothing, which should trigger the warning. Also several other places in the code, where `[[maybe_unused]]` is used.

Reply:
For some accelerator types the test is not called. I had compile errors. Therefore it was necessary.